storage/src/main/java/com/strategyobject/substrateclient/storage/StorageNMap.java

```diff
@@ -0,0 +1,159 @@
+ * @param <V> the type of the value.
+ */
+public interface StorageNMap<V> {
+    /**
+     * Returns the value by given key(s).
+     * @param keys the keys of the storage's entry.
+     * @return the value.
+     */
+    CompletableFuture<V> get(@NonNull Object... keys);
+
+    /**
+     * Returns the value by given key(s)
+     * starting at block hash given as the first parameter.
+     * @param block hash of the block
+     *              since that the value is requested.
+     * @param keys the keys of the storage's entry.
+     * @return the value.
+     */
+    CompletableFuture<V> at(@NonNull BlockHash block, @NonNull Object... keys);
+
+    /**
+     * Retrieves the storage hash.
+     * @param keys the keys of the storage's entry.
+     * @return the hash.
+     */
+    CompletableFuture<Hash> hash(@NonNull Object... keys);
+
+    /**
+     * Retrieves the storage hash
+     * starting at block hash given as the first parameter.
+     * @param block hash of the block
+     *              since that the hash is requested.
+     * @param keys the keys of the storage's entry.
+     * @return the hash.
+     */
+    CompletableFuture<Hash> hashAt(@NonNull BlockHash block, @NonNull Object... keys);
+
+    /**
+     * Retrieves the storage size.
+     * @param keys the keys of the storage's entry.
+     * @return the size of the entry.
+     */
+    CompletableFuture<Long> size(@NonNull Object... keys);
+
+    /**
+     * Retrieves the storage size
+     * starting at block hash given as the first parameter.
+     * @param block hash of the block
+     *              since that the size is requested.
+     * @param keys the keys of the storage's entry.
+     * @return the size of the entry.
+     */
+    CompletableFuture<Long> sizeAt(@NonNull BlockHash block, @NonNull Object... keys);
+
+    /**
+     * Returns collection of keys each item of which
+     * contains the rest keys of the entry.
+     * @param keys some keys of the storage's entry.
+     *             Note: each key must follow the other in the strict order as
+     *             they represented in the storage.
+     * @return collection of keys.
+     */
+    CompletableFuture<KeyCollection<V>> keys(@NonNull Object... keys);
+
+
+    /**
+     * Returns collection of keys each item of which
+     * contains the rest keys of the entry.
+     * @param block hash of the block
+     *              since that the keys are requested.
+     * @param keys some keys of the storage's entry.
+     *             Note: each key must follow the other in the strict order as
+     *             they represented in the storage.
+     * @return collection of keys.
+     */
+    CompletableFuture<KeyCollection<V>> keysAt(@NonNull BlockHash block, @NonNull Object... keys);
+
+    /**
+     * Returns iterator of collection of keys each item of which
+     * contains the rest keys of the entry.
+     * Each item of the iterator contains key collection of size that equals to page size
```

Comment:
Each item of the iterator contains a key collection with size equal to the page's size